src/components/Players/PlayersReducer.js

```diff
@@ -2,15 +2,27 @@ import { types } from './PlayersActionTypes';
 
 export const initialState = {
     players: [], /* [{ username, score }] */
+    newScores: [], /* [{ username, newScore }] */
     currentPlayer: null, /* { username } */
 };
 
 const playersReducer = (state = initialState, action) => {
+    
     switch (action.type) {
         case types.SET_PLAYERS:
-            return {...state, players: action.players };
+            return { ...state, players: action.players };
+        case types.SET_SCORES_FOR_ROUND:
+            return { 
+                ...state,
+                newScores: action.players.map(player => ({
```

Comment:
I think it looks nicer if you do the logic in the action and then just pass it to the reducer